Code/Framework/AzCore/AzCore/Serialization/Json/JsonSerializationSettings.h

```diff
@@ -36,6 +36,11 @@ namespace AZ
         //! any values in the container will be kept and not overwritten.
         //! Note that this does not apply to containers where elements have a fixed location such as smart pointers or AZStd::tuple.
         bool m_clearContainers = false;
+
+        //! If true, when a key already exists in the container, the value associated with it will be updated instead of creating a new
+        //! default object. If the key does not exist, it will be created. This essentially allows patching data in associative containers.
+        //! If false, existing keys in the container will be replaced.
+        bool m_updateAssociativeContainer = false;
```

Comment:
The behavior of the Json Serialization is to take the existing object it's provided and patch the information from the JSON document onto it. So if the given associated container has an existing member with the same name as in the JSON object then according to those rules the expected default here should be `true`. If this isn't desired than the user can remove the value that needs to be updated from the map, either manually, by setting `m_clearContainers` to `true` or by having the full object in the JSON object to effectively fully replace the existing object. Along that line of thought I even wonder if this should be an option and not just standard behavior? (More in the answer to question from @amzn-sj below.)

Reply:
Ok. I'll add sig-core to the review and wait for comments. If we're making this the standard behavior, then we'll need to update the existing MapSerializer test case to reflect this. I hope there isn't anything around that expects the old behavior.

Reply:
I think the default behavior should be to update keys with existing values in the container during JSON deserialization to a C++ object instead of silently not updating the container those keys.

If I were were to use the Json Deserializer I would (already have the container be cleared, but that is beside the point) expect any values at an existing key to be overridden.